cmake/modules/extensions.cmake

```diff
@@ -1559,6 +1559,66 @@ function(zephyr_syscall_header_ifdef feature_toggle)
   endif()
 endfunction()
 
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a warning/
+# fatal error message is printed (depends on REQUIRED flag).
+# Paths are relative to the current module's zephyr/blobs directory.
+#
+# Usage:
+#   zephyr_blobs_verify([PATHS path [paths...]]
+#                       [REQUIRED])
+#
+# Example:
+# zephyr_blobs_verify(REQUIRED)           # verify all blobs and fail on error
+# zephyr_blobs_verify(PATHS img/file.bin) # verify a single path
+function(zephyr_blobs_verify)
+  if(NOT DEFINED ZEPHYR_CURRENT_MODULE_NAME)
+    message(FATAL_ERROR "Unknown module name when calling ${CMAKE_CURRENT_FUNCTION}")
+  endif()
+
+  if(NOT WEST)
+    return()
+  endif()
+
+  cmake_parse_arguments(BLOBS_VERIFY "REQUIRED" "" "PATHS" ${ARGN})
+
+  execute_process(
+    COMMAND ${WEST} blobs list ${ZEPHYR_CURRENT_MODULE_NAME} --format "{status} {path}"
+    OUTPUT_VARIABLE BLOBS_LIST_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    COMMAND_ERROR_IS_FATAL ANY
+  )
+
+  if(${BLOBS_VERIFY_REQUIRED})
+    set(msg_lvl FATAL_ERROR)
+  else()
+    set(msg_lvl WARNING)
+  endif()
+
+  string(REPLACE "\n" ";" BLOBS_LIST ${BLOBS_LIST_OUTPUT})
+
+  if(DEFINED BLOBS_VERIFY_PATHS)
+    foreach(path ${BLOBS_VERIFY_PATHS})
+      message(VERBOSE "Verifying blob \"${path}\"")
+
+      # Each path that has a correct sha256 is prefixed with an A
+      if(NOT "A ${path}" IN_LIST BLOBS_LIST)
+        message(${msg_lvl} "Blob for path \"${path}\" isn't valid. Update with: west blobs fetch")
```

Comment:
Please inform users to just fetch blobs for the module which is failing.

No reason to fetch blobs for a module which may not be used by the user.